packages/next/shared/lib/router/utils/is-bot.ts

```diff
@@ -0,0 +1,5 @@
+export function isBot(userAgent: string): boolean {
+  return /Googlebot|Mediapartners-Google|AdsBot-Google|googleweblight|Storebot-Google|Google-PageRenderer|Bingbot|BingPreview|Slurp|DuckDuckBot|baiduspider|yandex|sogou|LinkedInBot|bitlybot|tumblr|vkShare|quora link preview|facebookexternalhit|facebookcatalog|Twitterbot|applebot|redditbot|Slackbot|Discordbot|WhatsApp|SkypeUriPreview|ia_archiver/i.test(
+    userAgent
```

Comment:
Now that this is shared with the client, can we reduce the size of this regex?  Perhaps “bot” will cover most of these cases. Something like

  /bot|google|bing|baidu|slurp|facebook|yandex|sogou|tumblr|vkshare|quora|whatsapp|skype|ia_archiver/i

Reply:
Possibly we could use a more minimal regex specifically for the client case since if we change this we will need to update it for the proxy as well

Reply:
I would like to keep client and server (and proxy) implementation the same

Reply:
Should we merge this as is, and then later update?

The best strategy for the client-specific RE would be to eliminate the bots from the list that
- don't match `/bot/`
- and which are known to not actually render the HTML (and hence would never actually execute on the client). Whatsapp/Skype would be candidates for this.

TBH I'm somewhat sceptical that there are legit bots that don't put `bot` somewhere into their UA, but I have no evidence :)

Reply:
Sounds good to me to land this as is and discuss tweaking the bot user agent regex separate.